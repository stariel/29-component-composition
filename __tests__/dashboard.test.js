import React from "react";
import renderer from "react-test-renderer";
import 'jest-enzyme';
import {Enzyme} from 'enzyme';
import { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import Dashboard from '../src/components/dashboard/dashboard.js';
import NoteCreateForm from '../src/components/note-create-form/note-create-form.js';


describe('<Dashboard/> (Enzyme Test)', () => {
  it('is alive at application start', () => {
    let app = shallow(<Dashboard/>);
    expect(app.find('h2').exists()).toBeTruthy();
  });
});

describe('<Dashboard/> (Snapshot Test)', () => {
  it('renders right', () => {
    const component = renderer.create(
      <Dashboard />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

  });

});

  describe('<NoteCreateForm/> (Enzyme Test)', () => {
    it('is alive at application start', () => {
      let app = shallow(<NoteCreateForm/>);
      expect(app.find('form').exists()).toBeTruthy();
    });
  });
  
  describe('<NoteCreateForm/> (Snapshot Test)', () => {
    it('renders right', () => {
      const component = renderer.create(
        <NoteCreateForm />
      );
  
      let tree = component.toJSON();
  
      expect(tree).toMatchSnapshot();
  
    });
});