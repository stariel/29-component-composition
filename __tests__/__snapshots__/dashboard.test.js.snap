// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> (Snapshot Test) renders right 1`] = `
Array [
  <h2>
    Notes App
  </h2>,
  <form
    onChange={[Function]}
    onSubmit={[Function]}
  >
    <input
      name="title"
      placeholder="Title"
      value=""
    />
    <input
      name="content"
      placeholder="Content"
      value=""
    />
    <label>
      <span>
        editing
      </span>
      <input
        name="editing"
        type="checkbox"
      />
    </label>
    <label>
      <span>
        completed
      </span>
      <input
        name="completed"
        type="checkbox"
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>,
  <ul />,
]
`;

exports[`<NoteCreateForm/> (Snapshot Test) renders right 1`] = `
<form
  onChange={[Function]}
  onSubmit={[Function]}
>
  <input
    name="title"
    placeholder="Title"
    value=""
  />
  <input
    name="content"
    placeholder="Content"
    value=""
  />
  <label>
    <span>
      editing
    </span>
    <input
      name="editing"
      type="checkbox"
    />
  </label>
  <label>
    <span>
      completed
    </span>
    <input
      name="completed"
      type="checkbox"
    />
  </label>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
